src/lib/parquet/tests/Flow/Parquet/Tests/Integration/IO/CompressionTest.php

```diff
@@ -152,4 +152,55 @@ public function test_writing_and_reading_file_with_uncompressed_compression() :
         self::assertFileExists($path);
         \unlink($path);
     }
+
+    public function test_writing_and_reading_file_with_zstd_compression() : void
+    {
+        if (!\extension_loaded('zstd')) {
```

Comment:
since zstd is not available by default what I would do is I would put in phpunit.xml.dist a [section](https://phpunit.de/manual/6.5/en/organizing-tests.html#organizing-tests.xml-configuration) for tests that depends on zstd. Next I would try to create another github workflow that would compile and install that extension just to run those tests - but if you don't feel like working on it, you can keep it as is and I will add the zstd testsutie later

Reply:
should the section be specific, like:
```xml
<testsuite name="zstd">
  <file>src/lib/parquet/tests/Flow/Parquet/Tests/Integration/IO/CompressionTest.php</file>
  <file>src/lib/parquet/tests/Flow/Parquet/Tests/Unit/ParquetFile/CodecTest.php</file>
</testsuite>
```

or more broad like the other sections, like:

```xml
<testsuite name="zstd">
  <directory>src/lib/**/**/**/**/Tests/Integration</directory>
  <directory>src/lib/**/**/**/**/Tests/Unit</directory>
</testsuite>
```

Reply:
I would call the testsuite `zstd-extension` and pull those tests into standalone files like: 

- `src/lib/parquet/tests/Flow/Parquet/Tests/Integration/IO/Compression/ZSTDTest.php`
- `src/lib/parquet/tests/Flow/Parquet/Tests/Unit/ParquetFile/Codec/ZSTDTest.php`

and make it specific, because the goal is to execute only those tests with `--testsuite=extension-zstd`

Reply:
On a second note, shall I group the tests (using [test-groups](https://docs.phpunit.de/en/10.5/attributes.html#test-groups)) so that only the specific tests are executed?

Reply:
> On a second note, shall I group the tests (using [test-groups](https://docs.phpunit.de/en/10.5/attributes.html#test-groups)) so that only the specific tests are executed?

Yes! That's much better, I totally forgot about test-groups, nice catch!